test-utils/keygen/src/main.rs

```diff
@@ -1,9 +1,102 @@
+extern crate clap;
 extern crate primitives;
+extern crate serde_json;
 
+use clap::{App, Arg, ArgMatches, SubCommand};
+use primitives::hash::hash_struct;
+use primitives::signature::{SecretKey, sign};
 use primitives::signature::get_keypair;
+use primitives::types::TransactionBody;
+use std::fs;
+use std::path::Path;
+use std::process;
+use primitives::types::SignedTransaction;
+
+fn generate_keys(matches: &ArgMatches) {
+    let private_key_path = matches
+        .value_of("output_file")
+        .map(|x| Path::new(x))
+        .unwrap();
+
+    let public_key_path = private_key_path.with_extension("pub");
+
+    if private_key_path.exists() {
+        eprintln!("Private key file already exists: {:?}", &private_key_path);
+        process::exit(1);
+    } else if public_key_path.exists() {
+        eprintln!("Public key file already exists: {:?}", &public_key_path);
+        process::exit(1);
+    }
 
-fn main() {
     let (public_key, secret_key) = get_keypair();
-    println!("Public key: {}", public_key);
-    println!("Secret key: {}", secret_key);
+    fs::write(public_key_path, public_key.to_string()).unwrap();
```

Comment:
how about storing them in one json file?
Parity stores key file like this:
```
cat testnet1/chaindata/keystore/UTC--2018-10-06T21-00-24.984883678Z--825ed86da564b88102850d6e7a589912586727b7
{"address":"825ed86da564b88102850d6e7a589912586727b7","crypto":{"cipher":"aes-128-ctr","ciphertext":"...","cipherparams":{"iv":"..."},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"..."},"mac":"..."},"id":"68fe436a-91c4-4aac-99c0-e151daba63fc","version":3}
```
this is default, includs public key in the filename

Reply:
sounds good